Good Coding Practices/src/js/scripts.js

```diff
@@ -1,44 +1,114 @@
 // piece object
-const piece = (function() {
+const piece = (function () {
   let el = null;
-  const init = function(el) {
+  const initPos = null;
+  const init = function (el) {
     this.el = el;
+    this.initPos = this.el.getBoundingClientRect();
   };
-  const moveDelta = function(dx, dy) {
+  const setPos = function (dx, dy) {
+    this.el.style.left = `${dx}px`;
+    this.el.style.top = `${dy}px`;
+  }
+  const moveDelta = function (dx, dy) {
     const pos = this.el.getBoundingClientRect();
-    this.el.style.left = `${pos.left + dx}px`;
-    this.el.style.top = `${pos.top + dy}px`;
+    let newDx = (pos.left + dx);
+    let newDy = (pos.top + dy);
+    if ((newDx < (window.innerWidth - 100)) && (newDx > 0)) {
+      if ((newDy < (window.innerHeight - 100)) && (newDy > -1)) {
+        this.setPos(newDx, newDy);
+
+      }
+    }
   };
+
   return {
     init,
-    moveDelta
+    moveDelta,
+    setPos
   };
 })();
+function resetDelta() {

```

Comment:
Mixing responsibilities.
All info available in `Piece` - why it's a separate external function?